src/test/java/microsoft/exchange/webservices/data/RecurringAppointmentDateParsingTest.java

```diff
@@ -0,0 +1,139 @@
+/**************************************************************************
+ Exchange Web Services Java API
+ Copyright (c) Microsoft Corporation
+ All rights reserved.
+ MIT License
+ Permission is hereby granted, free of charge, to any person obtaining a copy of this software and associated documentation files (the ""Software""), to deal in the Software without restriction, including without limitation the rights to use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the Software, and to permit persons to whom the Software is furnished to do so, subject to the following conditions:
+ The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.
+ THE SOFTWARE IS PROVIDED *AS IS*, WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ **************************************************************************/
+package microsoft.exchange.webservices.data;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.GregorianCalendar;
+import java.util.TimeZone;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.JUnit4;
+
+@RunWith(JUnit4.class)
+public class RecurringAppointmentDateParsingTest extends BaseTest {
+
+  /**
+   * Parsing the api recurrence date-String, comparing old and new parsing code
+   * 
+   * @throws ParseException
+   */
+  @Test
+  public void parseAPIRecurrenceDateFormat() throws ParseException {
+    String dateString = "2015-01-08Z";
+    Date newParsing = exchangeServiceBaseMock.convertDateTimeStringToDate(dateString, false);
+    Date oldParsing = convertStartDateToUnspecifiedDateTime(dateString);
+    assertEquals(oldParsing, newParsing);
+  }
+
+  @Test
+  public void parseLocalPattern2NoDefault() {
+    String dateString = "2015-01-08T10:11:12"; // "yyyy-MM-dd'T'HH:mm:ss"
+    Date parsed = exchangeServiceBaseMock.convertDateTimeStringToDate(dateString, false);
+    Calendar calendar = new GregorianCalendar();
+    calendar.setTime(parsed);
+    assertEquals(2015, calendar.get(Calendar.YEAR));
+    assertEquals(0, calendar.get(Calendar.MONTH));

```

Comment:
http://stackoverflow.com/questions/344380/why-is-january-month-0-in-java-calendar